fs/config/config.go

```diff
@@ -126,3 +128,34 @@ type BackgroundFetchConfig struct {
 	// fetcher emits metrics
 	EmitMetricPeriodSec int64 `toml:"emit_metric_period_sec"`
 }
+
+type RetryableHTTPClientConfig struct {
+	// TimeoutConfig represents the settings for timeout at various points in a request lifecycle in a retryable http client.
+	TimeoutConfig `toml:"timeout"`
+	// RetryConfig represents the settings for retries in a retryable http client.
+	RetryConfig `toml:"retry"`
+}
+
+type TimeoutConfig struct {
+	// DialTimeout is the maximum duration that connection can take before a request attempt is timed out.
+	DialTimeout int64 `toml:"dial_timeout_msec"`
+	// ResponseHeaderTimeout is the maximum duration waiting for response headers before a request attempt is timed out.
+	// This starts after the entire request body is uploaded to the remote endpoint and stops when the request headers
+	// are fully read. It does not include reading the body.
+	ResponseHeaderTimeout int64 `toml:"response_header_timeout_msec"`
+	// RequestTimeout is the maximum duration before the entire request attempt is timed out. This starts when the
+	// client starts the connection attempt and ends when the entire response body is read.
+	RequestTimeout int64 `toml:"request_timeout_msec"`
```

Comment:
Can we get "Msec" appended to these struct members, similar to the RetryConfig?

Reply:
Sure.

Reply:
Even better: Is it possible to do something with go-toml such that we can actually represent these as durations and do the conversion while serializing/deserializing?

Reply:
So apparently v1 of `go-toml` supports serializing/deserializing `time.Duration`s but v2 of the package does not - https://github.com/pelletier/go-toml/issues/767. We have a couple options:

1. Create a custom `time.Duration` type that implements `UnmarshalText` - this way we can be compatible with v2 if we ever decide to upgrade - example: https://github.com/pelletier/go-toml/issues/767#issuecomment-1121299700
2. Stick to using v1 with built-in `time.Duration` support
3. Stick to using `int64` to represent duration's (what we currently do). This would make it the responsibility of the users of the config to handle converting them to `time.Duration`s.